### ISSUE TITLE
How to push custom events against event loop?

### ISSUE THREAD

**Issue by wu000227** [PistonDevelopers/piston#1322]
I am working with `GlutinWindow`, is there any way to push a custom `GenericEvent` against the main event loop? There is a `create_proxy()` function seems to be useful but I don't seem to have access to it.

A following up question, is it possible to do the window rendering and event handling in two separate threads communicate with each other with a `channel`?

The scenario is, my game logic also has its own event loop and multiple threads running in the background that may raise events at any time and need to be handled immediately. It accepts/produces inputs/outputs via `channel`s.

The problem is, it produce outputs not only when receiving inputs (background threads may also raise events that leads to outputs at any time).

What I am struggling with is to receive outputs from the game logic then render the screen and to handle piston events at the same time. What would be the ideal way to do this with piston?

**Comment by bvssvni** [comment]
You can create your own custom events here: https://docs.rs/pistoncore-input/0.28.0/input/enum.Event.html#variant.Custom

**Comment by wu000227** [comment]
How do I send it to the event loop? I have multiple threads that raise different custom events at any time, and cannot be refactored to use only main thread. Is there any mechanism like `channel`s or so I can use before the new `create_proxy` api is supported?

**Comment by bvssvni** [comment]
You're right! There is design mistake here. `Custom` should be moved to `Input`.
